File: src/main/数组/t118/Generate.java
package 数组.t118;

import java.util.ArrayList;
import java.util.List;

/**
 * @auther: sotowang
 * @date: 2019/12/06 19:48
 */
public class Generate {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        if (numRows == 0) {
            return res;
        }
        int[][] array = new int[numRows][numRows];
        List<Integer> list = null;
        for (int i = 0; i < numRows; i++) {
            array[i][0] = 1;
            list = new ArrayList<>();
            list.add(array[i][0]);
            for (int j = 1; j <= i; j++) {
                array[i][j] = array[i - 1][j - 1] + array[i - 1][j];
                list.add(array[i][j]);
            }
            res.add(list);
        }
        return res;
    }

    public static void main(String[] args) {
        System.out.println(new Generate().generate(5));
    }
}

File: src/main/数组/t118/generate.go
package t118

/**
https://leetcode-cn.com/problems/pascals-triangle/solution/go-shuang-bai-by-ba-xiang-6/
*/
func generate(numRows int) [][]int {
	if numRows == 0 {
		return [][]int{}
	}
	var res [][]int
	res = append(res, []int{1})
	for i := 1; i < numRows; i++ {
		m := []int{0}
		m = append(m, res[i-1]...)
		for j := 0; j < len(m)-1; j++ {
			m[j] += m[j+1]
		}
		res = append(res, m)
	}
	return res
}
